File: packages/csharp/src/Anip.Core/InputResolution.cs
using System.Text.Json.Serialization;

namespace Anip.Core;

public class InputResolution
{
    // Mode is required, but typed as nullable to distinguish "missing" from
    // "default enum value". Validate() rejects null mode at the cross-field check.
    [JsonPropertyName("mode")] public ResolutionMode? Mode { get; set; }
    [JsonPropertyName("resolver_ref")] public string? ResolverRef { get; set; }
    [JsonPropertyName("on_missing")] public ResolutionBehavior? OnMissing { get; set; }
    [JsonPropertyName("on_ambiguous")] public ResolutionBehavior? OnAmbiguous { get; set; }
    [JsonPropertyName("on_unresolved")] public ResolutionBehavior? OnUnresolved { get; set; }
}

File: packages/java/anip-core/src/main/java/dev/anip/core/InputResolution.java
package dev.anip.core;

public record InputResolution(
        ResolutionMode mode,
        String resolverRef,
        ResolutionBehavior onMissing,
        ResolutionBehavior onAmbiguous,
        ResolutionBehavior onUnresolved
) {
    public InputResolution {
        if (mode == null) {
            throw new IllegalArgumentException("resolution.mode is required");
        }
    }
}
